prog.go

```diff
@@ -58,7 +58,13 @@ type ProgramSpec struct {
 	Name string
 
 	// Type determines at which hook in the kernel a program will run.
-	Type       ProgramType
+	Type ProgramType
+
+	// AttachType of the program, needed to differentiate allowed context
+	// accesses in some newer program types like CGroupSockAddr. Should be
+	// set to AttachNone if not required.
```

Comment:
I think this is not necessary, since AttachNone is the default.

Reply:
True, it's a bit redundant. I copied this from the docstring on `type AttachType`, will remove it here.